### PR TITLE
ci: update k3d version

### PR DESCRIPTION
**Pull request by lahabana**
### Summary

- Update to latest k3d version
- Run some tasks in parallel
- Rely on previously built docker images to speed up the build in circleci
- Start clusters in parallel in e2e tests
- Fix eksctl install by using our own script

### Documentation

- [x] CI change only no docs 

### Testing

- [x] CI change implies testing 

### Backwards compatibility

- [x] Not changing any code 
- [x] Add `backport-to-stable` label if the code is backwards compatible. Otherwise, list breaking changes.

### CODE REVIEW

**Comment by lahabana** [comment]
Please also double check your local setup. I've checked myself but I think some people have different setups.

File: mk/k3d.mk [hunk 1]
```diff
@@ -1,28 +1,30 @@
...
-		  	--no-lb --no-hostip \
-		  	--network kind \
-		  	--port "$(PORT_PREFIX)80-$(PORT_PREFIX)89:30080-30089" \
-		  	--timeout 120s && \
-		until \
-			KUBECONFIG=$(KIND_KUBECONFIG) kubectl wait -n kube-system --timeout=5s --for condition=Ready --all pods ; \
-		do echo "Waiting for the cluster to come up" && sleep 1; done )
+k3d/start: ${KIND_KUBECONFIG_DIR} k3d/network/create
+	@KUBECONFIG=$(KIND_KUBECONFIG)  \
+		 k3d cluster create "$(KIND_CLUSTER_NAME)" \
```
Comment (jpeach) [inline mk/k3d.mk:19]: Should you use `$(K3D_PATH)` consistently? Or, if depending on the tools dir being in the path (we do that all the time) seems OK here, remove `${K3D_PATH)`  variable.

**Review by jpeach — APPROVED** [review]
This worked OK on my dev system.

File: mk/k3d.mk (hunk 1 above)
Comment (lahabana) [reply mk/k3d.mk:19]: I think either way is good and we're not really coherent anywhere in all the makefiles so going to keep the simplest syntax (using k3d directly and getting rid of K3D_PATH)

**Pull request merged**